### PR TITLE
Fix OPC DA reconnect recovery after disconnect (Fixes #101)

### LINKED ISSUE #101: OPC DA reconnect does not recover after disconnect
**Description**
OPC DA reconnect currently does not reliably recover after a connection loss in UA Edge Translator.

**Problem statement**
When an OPC DA connection is interrupted, the reconnect path does not consistently restore a working connection. In affected cases, the driver remains disconnected (or repeatedly retries without successful recovery), requiring manual intervention.

**Observed impact**
1. DA assets can remain unavailable after transient outages.
2. Runtime may continue retry loops without restoring normal read/write behavior.
3. Operational reliability is reduced for OPC Classic DA integrations.

**Proposed scope**
1. Fix reconnect logic so DA assets recover automatically after disconnects.
2. Handle endpoint parsing robustly (including URI-style values and host/port variants).
3. Make disconnect/reconnect cleanup resilient when underlying OPC/COM state is partially broken.
4. Improve reconnect diagnostics/logging for troubleshooting.
5. Add/adjust tests for reconnect success/failure branches and endpoint edge cases.

**Acceptance criteria**
1. After a transient disconnect, DA reconnect succeeds without manual restart in supported scenarios.
2. Invalid endpoint formats fail gracefully and follow retry policy.
3. Reconnect behavior is covered by updated automated tests.
4. Logs clearly show reconnect attempts, failures, and successful recovery.

**Implementation note**
I already have a candidate fix prepared and will open a PR linked to this issue for review.

### PR DESCRIPTION
**Pull request by binarystorm**
**Description**
This PR fixes OPC DA reconnect behavior so DA assets can recover after transient disconnects without requiring manual restart.

Fixes #101

**Problem**
OPC DA reconnect did not reliably restore connectivity after connection loss.
In some cases, endpoint parsing and disconnect/reconnect sequencing caused repeated failed retries and left assets disconnected.

**What changed**

1. Improved reconnect endpoint handling in UANodeManager.cs:
- supports URI-style endpoints (for example opc.da://host/...)
- keeps backward-compatible host[:port] parsing
- handles empty/invalid endpoint values gracefully and schedules retry
2. Hardened OPC DA disconnect/reconnect behavior in OPCDAAsset.cs:
- safer cleanup when COM/DCOM state is partially broken
- disconnect path now tolerates cleanup exceptions and still resets internal state
- improved structured logging for connect/disconnect/read/write paths
3. Updated OPCDA project platform handling in OPCDA.csproj:
- explicit platform target conditions for x86 vs non-x86 builds in this driver project context

### CODE REVIEW

File: ProtocolDrivers/OPCDA/OPCDA.csproj [hunk 1]
```diff
@@ -3,7 +3,8 @@
   <PropertyGroup>
     <TargetFramework>net10.0-windows</TargetFramework>
     <Platforms>x86</Platforms>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget Condition="'$(Platform)' == 'x86'">x86</PlatformTarget>
+    <PlatformTarget Condition="'$(Platform)' != 'x86'">x64</PlatformTarget>
```
Comment (barnstee) [inline ProtocolDrivers/OPCDA/OPCDA.csproj]: Building this for x64 doesn't make sense. It is x86 only.

**Review by barnstee — DISMISSED** [review]

File: ProtocolDrivers/OPCDA/OPCDA.csproj (hunk 1 above)
Comment (barnstee) [reply ProtocolDrivers/OPCDA/OPCDA.csproj]: Actually, see my other fix for this, PlatformTarget must be removed entirely.

Comment (binarystorm) [reply ProtocolDrivers/OPCDA/OPCDA.csproj]: Sorry for this. Fixed in ec20bd7.

File: UAServer/UANodeManager.cs [hunk 2]
```diff
@@ -2027,26 +2027,52 @@ private void TryReconnect(string assetId, IAsset asset)
...
+                    if (!uri.IsDefaultPort)
+                    {
+                        port = uri.Port;
+                    }
+                }
+                else
+                {
+                    // Backward-compatible fallback for host[:port] endpoints.
+                    string[] endpointParts = remoteEndpoint.Split(':');
+                    host = endpointParts.Length > 0 ? endpointParts[0] : null;
```
[the remark this answers is not in the document]
Comment (binarystorm) [reply UAServer/UANodeManager.cs:2290]: Fixed in ec20bd7.

File: ProtocolDrivers/OPCDA/OPCDAAsset.cs [hunk 3]
```diff
@@ -65,22 +65,34 @@ public void Disconnect()
...
-
-                _items.Clear();
-                Log.Logger.Information("Disconnected from OPC DA server");
-            }
-            catch (Exception ex)
-            {
-                Log.Logger.Error($"Error disconnecting from OPC DA server: {ex.Message}");
             }
+
+            Log.Logger.Information("Disconnected from OPC DA server");
```
[the remark this answers is not in the document]
Comment (binarystorm) [reply ProtocolDrivers/OPCDA/OPCDAAsset.cs]: Fixed in ec20bd7.

**Review by barnstee — DISMISSED** [review]

**Comment by barnstee** [comment]
PR is approved. Merge from main and we should be good to go.

**Comment by binarystorm** [comment]
> PR is approved. Merge from main and we should be good to go.

merged from main c0ea6b6.

**Review by barnstee — APPROVED** [review]

**Comment by barnstee** [comment]
@binarystorm, what's your email address? I want to send you a little token of my appreciation since you were the first external person who made a protocol driver contribution.

File: ProtocolDrivers/OPCDA/OPCDAAsset.cs [hunk 4]
```diff
@@ -49,37 +49,60 @@
...
-                    _group = null;
                 }
+            }
+            catch (Exception ex)
+            {
+                // A reconnect often begins after DCOM has already dropped the
+                // underlying connection. Cleanup must still continue.
+                Log.Logger.Debug(ex, "OPC DA group removal during disconnect failed.");
+                cleanupFailed = true;
+            }
```
[the remark this answers is not in the document]
Comment (barnstee) [reply ProtocolDrivers/OPCDA/OPCDAAsset.cs:78]: Rejected.
